### PR TITLE
fix(TDI-40170): Add HANA support to tJDBCSCDELT

### PR DESCRIPTION
**Pull request by 195858**
Adding support for SAP Hana.
Amended the begin template to conditionally include SAP Hana specific template if the appropriate mapping is selected.
Added a SAP HANA specific javajet template,

**What is the current behavior?** (You can also link to an open issue here)
When HANA mapping is selected the component generates SQL that is incompatible with SAP HANA.
See - TDI-40170
**What is the new behavior?**
Adding support for SAP Hana.
Amended the begin template to conditionally include SAP Hana specific template if the appropriate mapping is selected.
Added a SAP HANA specific javajet template,

When HANA is selected as mapping the correct SQL is generated. 

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.components.localprovider/components/tJDBCSCDELT/sap_hana.javajet [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        	whereClause.add("( d." + strFieldName + " is not null AND t." + strFieldName + " is null )");
+       	}
+        whereClause.add("( d." + strFieldName + " <> t." + strFieldName+")");
+    }
+    selectColumnString = StringUtils.join(columnNames.toArray(), ", t.");
+    columnNameString = StringUtils.join(columnNames.toArray(), ", ");
+    String setColumnsString = StringUtils.join(setColumns.toArray(), ", ");
+    String whereClauseString = StringUtils.join(whereClause.toArray(), " OR ");
+    %>
+    String updateForType1_<%=cid%> = "UPDATE d SET <%=setColumnsString%> FROM <%=text_util.wrap("tableName", cid)%> d, <%=text_util.wrap("sourceTable", cid)%> t WHERE <%=joinClauseString%> AND (<%=whereClauseString%>)";
```
Comment (wwang-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tJDBCSCDELT/sap_hana.javajet]: Please see the document : 
https://help.sap.com/viewer/4fe29514fd584807ac9f2a04f6754767/2.0.00/en-US/20ff268675191014964add3d17700909.html

advice the pattern like this : 
```
UPDATE T A SET VAL = B.VAR FROM T A, T2 B WHERE A.KEY = B.KEY;
```
now your pattern is : 
```
UPDATE A SET VAL = B.VAR FROM T A, T2 B WHERE A.KEY = B.KEY;
```
maybe your pattern is also right, but best to follow the sap document and we don't have test env to validate it now

File: main/plugins/org.talend.designer.components.localprovider/components/tJDBCSCDELT/sap_hana.javajet [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        	whereClause.add("( d." + strFieldName + " is null AND t." + strFieldName + " is not null )");
+        	whereClause.add("( d." + strFieldName + " is not null AND t." + strFieldName + " is null )");
+     	}
+        whereClause.add("( d." + strFieldName + " <> t." + strFieldName+")");
+    }
+    String whereClauseString = StringUtils.join(whereClause.toArray(), " OR ");
+    %>        
+    String changeDateTime_<%=cid%> = (new java.sql.Timestamp((Long)start_Hash.get("<%=cid %>"))).toString();
+	
+    String updateForType2_<%=cid %> = "UPDATE d SET <%=text_util.wrap(endDateField)%> = '" + (changeDateTime_<%=cid%>) + "'<%if (isEnableActive) {%>, <%=text_util.wrap(activeField)%> = 0<%}%> FROM  <%=text_util.wrap("tableName", cid)%> d, <%=text_util.wrap("sourceTable", cid)%> t WHERE <%=joinClauseString%> AND (<%=whereClauseString%>) AND d.<%=text_util.wrap(endDateField)%> IS NULL";
```
Comment (wwang-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tJDBCSCDELT/sap_hana.javajet]: the same with the comment above

File: main/plugins/org.talend.designer.components.localprovider/components/tJDBCSCDELT/sap_hana.javajet [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+     	}
+        whereClause.add("( d." + strFieldName + " <> t." + strFieldName+")");
+    }
+    String whereClauseString = StringUtils.join(whereClause.toArray(), " OR ");
+    %>        
+    String changeDateTime_<%=cid%> = (new java.sql.Timestamp((Long)start_Hash.get("<%=cid %>"))).toString();
+	
+    String updateForType2_<%=cid %> = "UPDATE d SET <%=text_util.wrap(endDateField)%> = '" + (changeDateTime_<%=cid%>) + "'<%if (isEnableActive) {%>, <%=text_util.wrap(activeField)%> = 0<%}%> FROM  <%=text_util.wrap("tableName", cid)%> d, <%=text_util.wrap("sourceTable", cid)%> t WHERE <%=joinClauseString%> AND (<%=whereClauseString%>) AND d.<%=text_util.wrap(endDateField)%> IS NULL"; 
+    <%
+    if(isEnableDebug) {
```
Comment (wwang-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tJDBCSCDELT/sap_hana.javajet]: duplicated code, should remove one

**Comment by 195858** [comment]
Hi @wwang-talend 

Have amended the template as suggested and tested against SAP Hana.
Works with the change suggested.

Rgds

Swen

**Pull request merged**